### PR TITLE
Pantry class missing functionality, calculateIngredCost, testing added

### PR DESCRIPTION
**Pull request by b-stein**
Type of change made:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling- no new features

Detailed Description:
Adds calculateIngredCost on Recipe class, will display on Instructions module under ingredients, the estimated cost of all ingredients for the given recipe. Happy and sad tests added too.

Why is this change required? What problem does it solve?
Finishes the Recipe missing functionality

Were there any challenges that arose while implementing this feature? If so, how were they addressed?

Files modified:
- [ ] index.html
- [ ] styles.css
- [x] scripts.js

Other files:

### CODE REVIEW

**Comment by KyleWong2510** [comment]
Looks good.  Thanks!

File: src/recipe.js [hunk 1]
```diff
@@ -4,13 +4,19 @@ class Recipe {
...
-    //   ingredientData.find(ingredient => ingredient === i);
-    // });
-  }
+		this.tags = recipe.tags;
+		this.instructions = recipe.instructions;
+	}
+	
+  calculateIngredCost(dataset) {
+		const totalCost = this.ingredients.reduce((acc, ingred) => {
+			const itemCost = dataset.find(i => i.id === ingred.id).estimatedCostInCents;
```
Comment (KyleWong2510) [inline src/recipe.js:13]: I like how you did this calculation.  i was thinking a map then reduce the mapped but i actually think this is less code and less loops.  nice work!

File: src/scripts.js [hunk 2]
```diff
@@ -32,9 +32,9 @@ let recipeData;
 let ingredientsData;
 
 let menuOpen = false;
-let recipes = [];
 let user;
 let pantry;
+let allRecipes = [];
```
Comment (KyleWong2510) [inline src/scripts.js:37]: does this just take the place of the previous recipes array?

**Comment by MarchCorbin** [comment]
Good work now we can calculate the costs of all those foods!

File: src/recipe.js (hunk 1 above)
Comment (b-stein) [reply src/recipe.js:13]: thanks! 🕶️

File: src/scripts.js (hunk 2 above)
Comment (b-stein) [reply src/scripts.js:37]: yea, in my head this name made more sense than "recipes", since so many singular "recipe" variables are used later

**Pull request merged**